postgres/src/tokio/test_utils.rs

```diff
@@ -0,0 +1,254 @@
+    ) -> Result<TableId, tokio_postgres::Error> {
+        let columns_str = columns
+            .iter()
+            .map(|(name, typ)| format!("{} {}", name, typ.to_string()))
+            .collect::<Vec<_>>()
+            .join(", ");
+
+        let create_table_query = format!(
+            "CREATE TABLE {} (id BIGSERIAL PRIMARY KEY, {})",
+            table_name.as_quoted_identifier(),
+            columns_str
+        );
+        self.client.execute(&create_table_query, &[]).await?;
+
+        // Get the OID of the newly created table
+        let row = self
+            .client
+            .query_one(
+                "SELECT c.oid FROM pg_class c JOIN pg_namespace n ON n.oid = c.relnamespace \
+            WHERE n.nspname = $1 AND c.relname = $2",
+                &[&table_name.schema, &table_name.name],
+            )
+            .await?;
+
+        let table_id: TableId = row.get(0);
+
+        Ok(table_id)
+    }
+
+    /// Inserts values into the specified table.
+    pub async fn insert_values(
+        &self,
+        table_name: TableName,
+        columns: &[&str],
+        values: &[&(dyn tokio_postgres::types::ToSql + Sync)],
+    ) -> Result<u64, tokio_postgres::Error> {
+        let columns_str = columns.join(", ");
+        let placeholders: Vec<String> = (1..=values.len()).map(|i| format!("${}", i)).collect();
+        let placeholders_str = placeholders.join(", ");
+
+        let insert_query = format!(
+            "INSERT INTO {} ({}) VALUES ({})",
+            table_name.as_quoted_identifier(),
+            columns_str,
+            placeholders_str
+        );
+
+        self.client.execute(&insert_query, values).await
+    }
+
+    /// Updates all rows in the specified table with the given values.
+    pub async fn update_values(
+        &self,
+        table_name: TableName,
+        columns: &[&str],
+        values: &[&str],
+    ) -> Result<u64, tokio_postgres::Error> {
+        let set_clauses: Vec<String> = columns
+            .iter()
+            .zip(values.iter())
+            .map(|(col, val)| format!("{} = {}", col, val))
+            .collect();
+        let set_clause = set_clauses.join(", ");
+
+        let update_query = format!(
+            "UPDATE {} SET {}",
+            table_name.as_quoted_identifier(),
+            set_clause
+        );
+
+        self.client.execute(&update_query, &[]).await
+    }
+
+    /// Queries rows from a single column of a table.
+    pub async fn query_table<T>(
+        &self,
+        table_name: &TableName,
+        column: &str,
+        where_clause: Option<&str>,
+    ) -> Result<Vec<T>, tokio_postgres::Error>
+    where
+        T: for<'a> tokio_postgres::types::FromSql<'a>,
+    {
+        let where_str = where_clause.map_or(String::new(), |w| format!(" WHERE {}", w));
+        let query = format!(
+            "SELECT {} FROM {}{}",
+            column,
+            table_name.as_quoted_identifier(),
+            where_str
+        );
```

Comment:
For queries like these we can use sqlx macros which are much nicer than manually formatting values into a string. Another minor point is about using uppercase in SQL: we use lowercase as a convention for SQL at supabase, but not a big deal here.

Reply:
Oh nice, good to know! I was wondering why everything was lowercase.

Reply:
Regarding sqlx, we can definitely unify things for later, I initially had a split database and everything just to not have any sqlx internals within `pg_replicate` but the ergonomics of `sqlx` are definitely better, will create a ticket.